src/EnergyPlus/HVACFan.cc

```diff
@@ -866,8 +868,8 @@ namespace HVACFan {
                             localFanTotEff = m_fanTotalEff;
                             locHiSpeedFanRunTimeFrac = locFanRunTimeFraction;
                             m_fanRunTimeFractionAtSpeed[0] += locHiSpeedFanRunTimeFrac;
-                            m_fanPower +=
-                                locHiSpeedFanRunTimeFrac * m_maxAirMassFlowRate * localPressureRise[mode] / (localFanTotEff * m_rhoAirStdInit);
+                            m_fanPower += max(
```

Comment:
I understand the intent here, but seeing all these spread out has me concerned about a performance impact.  Is there a more central location we can check for this and have less max calls later on?  Probably not but I feel I should ask.

Reply:
One central location for Fan:SystemModel check is [here](https://github.com/NREL/EnergyPlus/blob/167109939_Issue7365/src/EnergyPlus/HVACFan.cc#L976). But, will not work for "SpeedControlMethod::Continuous" due to the code [here](https://github.com/NREL/EnergyPlus/blob/167109939_Issue7365/src/EnergyPlus/HVACFan.cc#L939-L943) used to update fan outlet air condition.

The other possibility is to reset EMS fan pressure rise value [here](https://github.com/NREL/EnergyPlus/blob/167109939_Issue7365/src/EnergyPlus/HVACFan.cc#L751) but I do not know how significant would the performance impact be until we test it.

Reply:
@Myoldmopar @mjwitte Thinking about it, adding a check for the EMS pressure rise only is executed only when the EMS variable EMSFanPressureOverrideOn is true. This reduces the computation effort, but negative fan power will not be captured if it happen for other reasons, for instance when the Fan(FanNum).EMSAirMassFlowValue is negative.